### PR TITLE
clang - silence some pedantic warnings, fixes #742

### LINKED ISSUE #742: GNU extention warning
The following macros seem to use a GNU extension where the middle operand in `?:` is omitted:

https://github.com/CEED/libCEED/blob/caa531a793f3d84584e9b27c52e9992c3cc5ab43/include/ceed/ceed.h#L168-L176

I see warnings about this in MFEM when building with Apple clang with `-pedantic -std=c++11`. It will be nice if this use of a GNU extension can be avoided.

### PR DESCRIPTION
**Pull request by jeremylt**

### CODE REVIEW

**Comment by jeremylt** [comment]
Note: If Clang is being pedantic, it still doesn't like all of our use of VARGS, but that stuff is hidden away from MFEM's use of libCEED.

File: include/ceed/ceed.h [hunk 1]
```diff
@@ -180,10 +180,10 @@ CEED_EXTERN int CeedErrorImpl(Ceed, const char *, int, const char *, int,
 /// Use nonstandard ternary to convince the compiler/clang-tidy that this
 /// function never returns zero.
 #  define CeedError(ceed, ecode, ...)                                     \
-  (CeedErrorImpl((ceed), __FILE__, __LINE__, __func__, (ecode), __VA_ARGS__) ?: (ecode))
+  (CeedErrorImpl((ceed), __FILE__, __LINE__, __func__, (ecode), __VA_ARGS__) ? (ecode) : (ecode))
 #else
 #  define CeedError(ceed, ecode, ...)                                     \
-  CeedErrorImpl((ceed), __FILE__, __LINE__, __func__, (ecode), __VA_ARGS__) ?: (ecode)
+  CeedErrorImpl((ceed), __FILE__, __LINE__, __func__, (ecode), __VA_ARGS__) ? (ecode) : (ecode)
```
Comment (jedbrown) [inline include/ceed/ceed.h:186]: Once upon a time, the `__clang__` variant was to appease clang and the other was a standard ternary. I think the idea was to let `CeedErrorImpl` decide to return something different. What we have now is just comma operator; do the first thing and return the second.

File: interface/ceed-elemrestriction.c [hunk 2]
```diff
@@ -298,7 +298,7 @@ static struct CeedElemRestriction_private ceed_elemrestriction_none;
 /// @{
 
 /// Indicate that the stride is determined by the backend
-const CeedInt CEED_STRIDES_BACKEND[3] = {};
+const CeedInt CEED_STRIDES_BACKEND[3] = {0, 0, 0};
```
Comment (jedbrown) [inline interface/ceed-elemrestriction.c:301]: I think the standard says you can do it this way:
```suggestion
const CeedInt CEED_STRIDES_BACKEND[3] = {0};
```

**Review by jedbrown — APPROVED** [review]
What are the internal warnings?

**Comment by jeremylt** [comment]
last pedantic warning is
```
/home/jeremy/Dev/libCEED/backends/ceed-backend-weak.c:33:68: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
  int name(void) { return CeedRegister_Weak(__func__,num_prefixes, ## __VA_ARGS__); }
```

**Comment by jeremylt** [comment]
Ah, fixed by removing the token pasting. That wasn't needed.

**Comment by jeremylt** [comment]
Grrrrr, this again
```
libgcov profiling error:/var/lib/gitlab-runner/builds/N8GCsqus/0/libceed/libCEED/build/gallery/ceed-gallery-weak.gcda:overwriting an existing profile data with a different timestamp
```
It seems to happen most often when I push twice in succession

**Pull request merged**